src/Query/Wildcard.php

```diff
@@ -5,33 +5,56 @@
 /**
  * Wildcard query.
  *
- * @author Nicolas Ruflin <spam@ruflin.com>
- *
  * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html
  */
 class Wildcard extends AbstractQuery
 {
     /**
-     * Construct wildcard query.
-     *
-     * @param string $key   OPTIONAL Wildcard key
-     * @param string $value OPTIONAL Wildcard value
-     * @param float  $boost OPTIONAL Boost value (default = 1)
+     * Rewrite methods: @see https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-term-rewrite.html.
      */
-    public function __construct(string $key = '', ?string $value = null, float $boost = 1.0)
+    public const REWRITE_CONSTANT_SCORE = 'constant_score';
+    public const REWRITE_CONSTANT_SCORE_BOOLEAN = 'constant_score_boolean';
+    public const REWRITE_SCORING_BOOLEAN = 'scoring_boolean';
+
+    private $field;
+
+    public function __construct(string $field, string $value, float $boost = 1.0)
```

Comment:
I don't specifically remember it here but I know we had in the past reasons to not necesarily require all params in the constructur. The reasons was something like:

* The object is instantiated
* Passed to something else that sets the param and returned
* Then toArray is called

In summary, not all info is available on creation time. If this is the case here, not sure.

Reply:
It is quite weird that those constructors allows the creation of an "invalid" object, it leads to some weird anti-patterns IMHO.
What I'd like is to have queries (Terms, WildCard,..) being able to be constructed with the required parameters, and throw exceptions is case the required `field`is empty as an example.

In the scenario you described: it would be better to have the "something else" to be responsible to the creation of the object, with the right parameters. wdyt?

Reply:
Probably :-) I'm good with trying out what you suggest and if we get push back, we have someone that we can ask for the detailed use case!

Reply:
great! I'll continue working on this PR then :)
I'll open other PR for other classes, just to keep the diffs small